eng/pipelines/evaluate-changed-paths.sh

```diff
@@ -0,0 +1,168 @@
+'
+
+# Disable globbing in this bash script since we iterate over path patterns
+set -f
+
+usage()
+{
+  echo "Script that evaluates changed paths and emits an azure devops variable if the changes contained in the current HEAD against the difftarget meet the includepahts/excludepaths filters:"
+  echo "  --difftarget <value>       SHA or branch to diff against. (i.e: HEAD^1, origin/master, 0f4hd36, etc.)"
+  echo "  --excludepaths <value>     Escaped list of paths to exclude from diff separated by '+'. (i.e: 'src/libraries/*+'src/installer/*')"
+  echo "  --includepaths <value>     Escaped list of paths to include on diff separated by '+'. (i.e: 'src/libraries/System.Private.CoreLib/*')"
+  echo "  --subset                   Subset name for which we're evaluating in order to include it in logs"
+  echo "  --azurevariable            Name of azure devops variable to create if change meets filter criteria (Default: containschange)"
+  echo ""
+
+  echo "Arguments can also be passed in with a single hyphen."
+}
+
+source="${BASH_SOURCE[0]}"
+
+# resolve $source until the file is no longer a symlink
+while [[ -h "$source" ]]; do
+  scriptroot="$( cd -P "$( dirname "$source" )" && pwd )"
+  source="$(readlink "$source")"
+  # if $source was a relative symlink, we need to resolve it relative to the path where the
+  # symlink file was located
+  [[ $source != /* ]] && source="$scriptroot/$source"
+done
+
+scriptroot="$( cd -P "$( dirname "$source" )" && pwd )"
+eng_root=`cd -P "$scriptroot/.." && pwd`
+
+exclude_paths=()
+include_paths=()
+subset_name=''
+azure_variable="containschange"
+ci=true
+diff_target=""
+
+while [[ $# > 0 ]]; do
+  opt="$(echo "${1/#--/-}" | awk '{print tolower($0)}')"
+  case "$opt" in
+    -help|-h)
+      usage
+      exit 0
+      ;;
+    -difftarget)
+      diff_target=$2
+      shift
+      ;;
+    -excludepaths)
+      IFS='+' read -r -a tmp <<< $2
+      exclude_paths+=($tmp)
+      shift
+      ;;
+    -includepaths)
+      IFS='+' read -r -a tmp <<< $2
+      include_paths+=($tmp)
+      shift
+      ;;
+    -subset)
+      subset_name=$2
+      shift
+      ;;
+    -azurevariable)
+      azure_variable=$2
+      shift
+      ;;
+  esac
+
+  shift
+done
+
+. "$eng_root/common/pipeline-logging-functions.sh"
+
+# expected args
+# $@: filter string
+function runGitDiff {
+  local _filter=$@
+  echo ""
+  echo "git diff -M -C -b --ignore-cr-at-eol --ignore-space-at-eol --exit-code --quiet $diff_target -- $_filter"
+  git diff -M -C -b --ignore-cr-at-eol --ignore-space-at-eol --exit-code --quiet $diff_target -- $_filter
+  git_diff_exit_code=$?
+}
+
+# expected args
+# $@: filter string
+function printMatchedPaths {
+  local _subset=$subset_name
+  local _filter=$@
+  echo ""
+  echo "----- Matching files for $_subset -----"
+  git diff -M -C -b --ignore-cr-at-eol --ignore-space-at-eol --name-only $diff_target -- $_filter
```

Comment:
Consider taking out the preamble to emphasize the important difference between `printMatchedPaths` and `runGitDiff` (don't know about a good function name):

```sh
permissiveGitDiff() {
  git diff -M -C -b --ignore-cr-at-eol --ignore-space-at-eol "$@"
}
```